include/crm/services.h

```diff
@@ -170,112 +171,123 @@ typedef struct svc_action_s {
+GList *resources_list_agents(const char *standard, const char *provider);
 
-/**
+/*!
  * Get list of available standards
  *
- * \return a list of resource standards. The list items are char *. This list _must_
- *         be destroyed using g_list_free_full(list, free).
+ * \return List of resource standards as char * items.
+ * \note The caller is responsible for freeing the result using
+ *       g_list_free_full(list, free).
  */
-    GList *resources_list_standards(void);
+GList *resources_list_standards(void);
 
-/**
- * Does the given standard, provider, and agent describe a resource that can exist?
+/*!
+ * \brief Check whether a resource agent exists on the local host
  *
- * \param[in] standard  Which class of agent does the resource belong to?
- * \param[in] provider  What provides the agent (NULL for most standards)?
- * \param[in] agent     What is the name of the agent?
+ * \param[in] standard  Resource agent standard of agent to check
+ * \param[in] provider  Provider of agent to check (or NULL)
+ * \param[in] agent     Name of agent to check
  *
- * \return A boolean
+ * \return TRUE if agent exists locally, otherwise FALSE
  */
-    gboolean resources_agent_exists(const char *standard, const char *provider, const char *agent);
+gboolean resources_agent_exists(const char *standard, const char *provider,
+                                const char *agent);
 
-/**
+/*!
  * \brief Create a new resource action
  *
- * \param[in] name        Name of resource
- * \param[in] standard    Resource agent standard (ocf, lsb, etc.)
+ * \param[in] name        Name of resource that action is for
+ * \param[in] standard    Resource agent standard
  * \param[in] provider    Resource agent provider
  * \param[in] agent       Resource agent name
- * \param[in] action      action (start, stop, monitor, etc.)
+ * \param[in] action      Name of action to create
  * \param[in] interval_ms How often to repeat this action (if 0, execute once)
- * \param[in] timeout     Consider action failed if it does not complete in this many milliseconds
+ * \param[in] timeout     Error if not complete within this many milliseconds
  * \param[in] params      Action parameters
  *
- * \return newly allocated action instance
+ * \return Newly allocated action
  *
- * \post After the call, 'params' is owned, and later free'd by the svc_action_t result
+ * \note The returned result assumes ownership of \p params.
  * \note The caller is responsible for freeing the return value using
  *       services_action_free().
  */
 svc_action_t *resources_action_create(const char *name, const char *standard,
                                       const char *provider, const char *agent,
                                       const char *action, guint interval_ms,
-                                      int timeout /* ms */, GHashTable *params,
+                                      int timeout, GHashTable *params,
```

Comment:
Could change to `timeout_ms` for clarity; I'm on the fence since we're stuck with `svc_action_t:timeout` and it's nice to stay parallel with that.

Reply:
It should be changed, but the function definition should be changed to match, and I think it's out of scope of this PR -- it can wait until a "best practices" overhaul of libcrmservice